tools/bin/commands/dev

```diff
@@ -1,15 +1,102 @@
+    # generate a new user so that it will be asked to approve permissions delegation
+    local username="dev$RANDOM"
+
+    # clean cookie
+    rm -rf "$COOKIE1"
+    touch "$COOKIE1"
+
+    local syndesis_url="$(oc get route syndesis  --template={{.spec.host}})"
+    local openshift_ip_and_port="$(oc status | grep "In project" | grep  -Po --color=never '(?<=server https://).*' )"
+    # first invocation is needed just to get the redierect url and to define a csrf token
+    local redirect_url="$(curl --write-out "%{url_effective}\n" --location --silent --show-error --output /dev/null --insecure --cookie-jar  "$COOKIE1" -v "https://$syndesis_url" 2>/dev/null )"
+
+    # isolate required params
+    # split on question mark, so we are sure the first part is always the correct url
+    local login_url="$(echo $redirect_url | sed -e 's/?/\n/g' | head -n1)"
+    # split on question marks or ampersend, than filter on the specific param we are interested into
+    local then_url="$(echo $redirect_url | sed -e 's/[?&]/\n/g' | grep "then=" | sed -e 's/then=//' )"
+    local csrf=$(cat "$COOKIE1"  | grep -Po '(?<=\scsrf).*' | awk '{$1=$1;print}')
+
+    # login on OpenShift and save output to delegate_permissions_form location
+    local delegate_permissions_form="$(mktemp /tmp/syndesis_permissions.XXX.html)"
+    curl --referer "$redirect_url" --insecure --location  -H "Cookie: csrf=$csrf" --data-urlencode username="$username"  --data-urlencode password="developer" --data-ascii then="$then_url" --data-urlencode csrf="$csrf" --cookie-jar "$cookie2" -i -v -o $delegate_permissions_form "$login_url"  2>/dev/null
+
+    # move the freshly created "ssn" cookie entry back to the main cookie
+    tail -1 "$cookie2" >> "$COOKIE1"
+
+    local then_url=$(extract_and_escape "then" $delegate_permissions_form )
+    local csrf=$(extract_and_escape "csrf" $delegate_permissions_form )
+    local client_id=$(extract_and_escape "client_id" $delegate_permissions_form )
+    local user_name=$(extract_and_escape "user_name" $delegate_permissions_form )
+    local redirect_uri=$(extract_and_escape "redirect_uri" $delegate_permissions_form )
+
+    local scopes=$(grep -Po '(?<=name="scope").*' $delegate_permissions_form | grep -Po '(?<=value=")[^"]+')
+    local joined_scopes=""
+    
+    scopes=(${scopes// / })
+    for i in "${!scopes[@]}"
+    do
+        local encoded="$(dev::urlencode ${scopes[i]})"
+        joined_scopes="$joined_scopes --data scope=$encoded"
+    done
+
+    # approve permission delegation, weaving all the params in the GET parameters
+    
+    curl --insecure --location --cookie-jar "$COOKIE1" --cookie "$COOKIE1" -i -v --data "then=$then_url" --data "csrf=$csrf" --data "client_id=$client_id" --data "user_name=$user_name" --data "redirect_uri=$redirect_uri" $joined_scopes --data "approve=Allow+selected+permissions"  -H 'dnt: 1'  "https://$openshift_ip_and_port/oauth/authorize/approve" -o /dev/null 2>/dev/null
```

Comment:
@paoloantinori can't you use `--data-urlencode` here for the parameters and safe the custom encoding on your own ?